pkg/common/configs/configwatcher.go

```diff
@@ -111,16 +111,19 @@ func (cw *ConfigWatcher) Run() {
 		quit := make(chan bool)
 		go func() {
 			for {
+				select {
+				case <-quit:
+					<-cw.soloChan
+					ticker.Stop()
+					return
+				default:
+				}
 				select {
 				case <-ticker.C:
 					if !cw.runOnce() {
 						<-cw.soloChan
 						return
 					}
-				case <-quit:
-					<-cw.soloChan
-					ticker.Stop()
-					return
```

Comment:
This case should not be removed. If you are in this select we could wait for the ticker to expire (1 sec) before we exit to get triggered in the select above.
This means that if the quit channel is checked, no quit, we move into this select and wait. If the quit is triggered just after we go into this 2nd select we wait 1 sec. A repeated run of the test shows that not having the select adds about 10 seconds on 100 runs. In other words it occurs 1 in every 10 cases.
Setting up a priority select should always follow the double select pattern etc

Reply:
Thanks for the detailed analysis, got it.